lib/api/requests.ts

```diff
@@ -18,8 +24,10 @@ const OPTIONAL_FIELDS = Object.freeze([
   'queue_draft',
 ]);
 
-class SendEmailRequest {
-  constructor(opts) {
+export class SendEmailRequest {
+  message: Message;
+
+  constructor(opts: Record<string, unknown>) {
```

Comment:
We could type the `opts` too maybe. All the fields are known, so it will serve as documentation. Maybe something like this?


```
type Identifiers = {
  id?: string;
  email?: string;
};

type SendEmailRequestRequiredOptions = {
  to: string;
  identifiers: Identifiers;
};

type SendEmailRequestOptionalOptions = Partial<{
  message_data: Record<string, any>;
  transactional_message_id: string;
  headers: Record<string, any>;
  preheader: string;
  from: string;
  reply_to: string;
  bcc: string;
  subject: string;
  body: string;
  plaintext_body: string;
  amp_body: string;
  fake_bcc: string;
  disable_message_retention: boolean;
  send_to_unsubscribed: boolean;
  tracked: boolean;
  queue_draft: boolean;
}>;

type SendEmailRequestOptions = SendEmailRequestOptionalOptions & SendEmailRequestRequiredOptions;
```

In the constructor, we can then remove the loops and create the message object directly. It's a bit verbose, but it'll make the experience of using this library better.

```
  constructor(opts: SendEmailRequestOptions) {
    this.message = {
      to: opts.to,
      identifiers: opts.identifiers,
      attachments: {},
      message_data: opts.message_data,
      transactional_message_id: opts.transactional_message_id,
      headers: opts.headers || {},
      preheader: opts.preheader,
      from: opts.from,
      reply_to: opts.reply_to,
      bcc: opts.bcc,
      subject: opts.subject,
      body: opts.body,
      plaintext_body: opts.plaintext_body,
      amp_body: opts.amp_body,
      fake_bcc: opts.fake_bcc,
      disable_message_retention: opts.disable_message_retention,
      send_to_unsubscribed: opts.send_to_unsubscribed,
      tracked: opts.tracked,
      queue_draft: opts.queue_draft,
    };
  }
```

Reply:
My 2c: Identifiers type should be:
```typescript
type Identifiers = { id: string } | { email: string };
```

(Identifiers aren't optional, so no `id?`/`email?`: at the same time you can't pass both of them, only one is allowed.)

Reply:
Oh, good point, you're right!